packages/editor-ui/src/components/NodeCredentials.vue

```diff
@@ -543,8 +559,20 @@ export default defineComponent({
 		},
 
 		editCredential(credentialType: string): void {
-			const { id } = this.node.credentials[credentialType];
-			this.uiStore.openExistingCredential(id);
+			const credential = this.node.credentials?.[credentialType];
+
+			if (!credential?.id) {
+				this.showMessage({
+					title: this.$locale.baseText('nodeCredentials.editCredentialError.title'),
+					message: this.$locale.baseText('nodeCredentials.editCredentialError.message', {
+						interpolate: { credentialType },
+					}),
+					type: 'error',
+				});
+				return;
+			}
```

Comment:
Is this an expected situation that can happen? If yes, then this is good. If not, we can use `assert` to ensure the credential exists and it's fine to explode the app (i.e. then it's a bug somewhere if that happens)

Reply:
I am pretty sure this is not expected so agree assert is the way to go here. Will update.